python/cudf/cudf/tests/test_parquet.py

```diff
@@ -210,6 +210,35 @@ def test_parquet_reader_basic(parquet_file, columns, engine):
     assert_eq(expect, got, check_categorical=False)
 
 
+@pytest.mark.filterwarnings("ignore:Using CPU")
+@pytest.mark.parametrize("engine", ["cudf"])
+def test_parquet_reader_empty_pandas_dataframe(tmpdir, engine):
+    df = pd.DataFrame()
+
+    fname = tmpdir.join("test_pq_reader_empty_pandas_dataframe.parquet")
+
+    df.to_parquet(fname)
+    assert os.path.exists(fname)
+
+    expect = pd.read_parquet(fname)
+    got = cudf.read_parquet(fname, engine="cudf")
+
+    if len(expect) == 0:
+        expect = expect.reset_index(drop=True)
+        got = got.reset_index(drop=True)
+        if "col_category" in expect.columns:
+            expect["col_category"] = expect["col_category"].astype("category")
```

Comment:
We shouldn't need to check if the length is 0 since that's the whole point of this test. Additionally don't need to handle `category` columns.

Reply:
Also, ideally we shouldn't need to do a `reset_index` to match Pandas, but would be good to understand why we currently have to do so.

Reply:
Sounds good, I think I might have tried removing some of this and got an error. I'll try it out without and let you know if there's something that goes wrong with it!